flink/v1.17/flink/src/main/java/org/apache/iceberg/flink/util/FlinkAlterTableUtil.java

```diff
@@ -0,0 +1,176 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.iceberg.flink.util;
+
+import java.util.List;
+import org.apache.flink.table.catalog.Column;
+import org.apache.flink.table.catalog.TableChange;
+import org.apache.flink.table.catalog.UniqueConstraint;
+import org.apache.iceberg.UpdateProperties;
+import org.apache.iceberg.UpdateSchema;
+import org.apache.iceberg.flink.FlinkSchemaUtil;
+import org.apache.iceberg.relocated.com.google.common.base.Preconditions;
+import org.apache.iceberg.types.Type;
+
+public class FlinkAlterTableUtil {
+  private FlinkAlterTableUtil() {}
+
+  /**
+   * Applies a list of Flink table changes to an {@link UpdateSchema} operation.
+   *
+   * @param pendingUpdate an uncommitted UpdateSchema operation to configure
+   * @param schemaChanges a list of Flink table changes
+   */
+  public static void applySchemaChanges(
+      UpdateSchema pendingUpdate, List<TableChange> schemaChanges) {
+    for (TableChange change : schemaChanges) {
+      if (change instanceof TableChange.AddColumn) {
+        TableChange.AddColumn addColumn = (TableChange.AddColumn) change;
+        Column flinkColumn = addColumn.getColumn();
+        Preconditions.checkArgument(
+            FlinkCompatibilityUtil.isPhysicalColumn(flinkColumn),
+            "Adding computed columns is not supported yet: %s",
+            flinkColumn.getName());
+        Type icebergType = FlinkSchemaUtil.convert(flinkColumn.getDataType().getLogicalType());
+        pendingUpdate.addColumn(flinkColumn.getName(), icebergType);
+
+      } else if (change instanceof TableChange.ModifyColumn) {
+        TableChange.ModifyColumn modifyColumn = (TableChange.ModifyColumn) change;
+        applyModifyColumn(pendingUpdate, modifyColumn);
+
+      } else if (change instanceof TableChange.DropColumn) {
+        TableChange.DropColumn dropColumn = (TableChange.DropColumn) change;
+        pendingUpdate.deleteColumn(dropColumn.getColumnName());
+
+      } else if (change instanceof TableChange.AddWatermark) {
+        throw new UnsupportedOperationException("Adding watermark specs is not supported yet. ");
```

Comment:
this (the extra whitespace in the end) needs to be addressed.

Reply:
also the error msg doesn't follow the coding style. you can referring to `ArrowVectorAccessor` for examples. Please review all the error msgs in this class

E.g., it could be sth like `Unsupported schema change: adding watermark specs`